src/Microsoft.Fx.Portability/Analyzer/RequestAnalyzer.cs

```diff
@@ -60,13 +48,38 @@ public AnalyzeResponse AnalyzeRequest(AnalyzeRequest request, string submissionI
                 ? _analysisEngine.FindBreakingChangeSkippedAssemblies(targets, request.UserAssemblies, request.AssembliesToIgnore).ToList()
                 : new List<AssemblyInfo>();
 
+            var userAssemblies = new HashSet<string>(assemblyIdentities, StringComparer.OrdinalIgnoreCase);
+            var assembliesToRemove = new HashSet<string>();
+            var nugetPackages = new List<NuGetPackageInfo>();
+
+            // If the request contains the list of referenced NuGet packages (which it should if it comes from Visual Studio), find if there are supported versions for those packages.
+            // If the request does not contain the list of referenced NuGet packages (request comes from command line version of the tool), get package info for user assemblies and for missing assemblies.
+            // Also remove from analysis those user assemblies for which supported packages are found.
+            if (request.ReferencedNuGetPackages != null)
+            {
+                nugetPackages = _analysisEngine.GetNuGetPackagesInfo(request.ReferencedNuGetPackages, targets).ToList();
+            }
+            else
+            {
+                var nugetPackagesForUserAssemblies = _analysisEngine.GetNuGetPackagesInfoFromAssembly(assemblyIdentities, targets);
+                assembliesToRemove = new HashSet<string>(_analysisEngine.ComputeAssembliesToRemove(request.UserAssemblies, targets, nugetPackagesForUserAssemblies), StringComparer.OrdinalIgnoreCase);
+
+                var nugetPackagesForMissingAssemblies = _analysisEngine.GetNuGetPackagesInfoFromAssembly(missingUserAssemblies, targets);
+                nugetPackages = nugetPackagesForMissingAssemblies.Union(nugetPackagesForUserAssemblies).ToList();
+            }
+            nugetPackages.Sort(new NuGetPackageInfoComparer());
+
+            userAssemblies.RemoveWhere(a => assembliesToRemove.Contains(a));
```

Comment:
you can simplify this to `userAssemblies.RemoveWhere(assembliesToRemove.Contains)`

Reply:
Changed.